frontend/simple/js/state.js

```diff
@@ -57,6 +64,24 @@ const mutations = {
       mutations.addContract(state, contract)
     }
   },
+  deleteMail (state, hash) {
+    let mailbox = getMailbox(state)
+    if (mailbox) {
+      let index = state[ mailbox ].messages.findIndex(msg => msg.hash === hash)
+      if (index > -1) {
+        state[ mailbox ].messages.splice(index, 1)
+      }
```

Comment:
This `if` block can be collapsed from 3 lines into 1 line to conform to the style of the rest of the and Save The Lions™.

The function can be simplified from 7 lines down to 3 lines and fewer braces/blocks using JavaScript's flexible `&&` operator:

```js
let mailboxState = mailboxForUser(userId)
let index = mailboxState && mailboxState.messages.findIndex(msg => msg.hash === hash)
index && mailboxState.splice(index, 1)
```